trio_websocket/_impl.py

```diff
@@ -440,12 +442,15 @@ def __init__(self, stream, wsproto, *, path=None):
         self._stream = stream
         self._stream_lock = trio.StrictFIFOLock()
         self._wsproto = wsproto
-        self._bytes_message = b''
-        self._str_message = ''
+        self._message_parts = []  # type: List[bytes|str]
         self._reader_running = True
         self._path = path
         self._subprotocol = None
-        self._send_channel, self._recv_channel = trio.open_memory_channel(0)
+        # TODO changed channel size from 0 to 2 temporarily to enable
+        # test_read_messages_after_remote_close to pass. The channel size will
+        # become a configurable setting when #74 is fixed.
+        self._send_channel, self._recv_channel = trio.open_memory_channel(
+            message_queue_size_do_not_use)
```

Comment:
I suggest removing this change, and defer adding test_read_messages_after_remote_close until channel size configuration is implemented.  The non-zero size is not needed to fix #74.

So i.e. this PR becomes merely a fix for #74.  In addition to test_no_messages_after_local_close, please add the test from #77.

Reply:
I'll go for the first suggestion: 

* Set channel size back to zero.
* Add the `skip` decorator to the test that requires non-zero channel size.
* Merge your test from #77.